collector/streaming.go

```diff
@@ -297,18 +297,23 @@ func (nc *channelsCollector) Describe(ch chan<- *prometheus.Desc) {
 func (nc *channelsCollector) Collect(ch chan<- prometheus.Metric) {
 	for _, server := range nc.servers {
 		var resp Channelsz
+		var serverResp StreamingServerz
+		if err := getMetricURL(nc.httpClient, server.URL, &resp); err != nil {
```

Comment:
why is this extra `getMetricURL` added? seems it is doing the same as  the call below?

Reply:
Hmm, that's the wrong way to do this. 

How would you suggest getting the server role into the Collect function here? This function is called with an array of servers via channelsCollector, so for each server I need to actually get a single metric from a _different_ URL.

1. Use some URL parsing to substitute `channelsz` with `serversz` and get one additional data point from the prior role?
2. If both `serversz` and `channelsz` are collected, run the former first and have it somehow return the role to apply to all of the latter. (This seems messy because both collectors are called with arrays of servers, not individual servers.)
3. ???

Reply:
I implemented in my new commit option #1, seeing no way without a major refactor to implement #2